salt/states/pkg.py

```diff
@@ -3629,3 +3629,313 @@ def mod_beacon(name, **kwargs):
             ),
             "result": False,
         }
+
+
+def held(name, version=None, pkgs=None, replace=False, **kwargs):
+    """
+    Set package in 'hold' state, meaning it will not be changed.
```

Comment:
This new function also needs a `.. versionadded:: Phosphorus` docs marker

Reply:
If you want this to go in for Phosphorus, we need to have versionadded. But it's 3005 now so:
```
.. versionadded:: 3005
```